tests/src/Kernel/PagerTest.php

```diff
@@ -0,0 +1,264 @@
+        ['url' => '?page=0', 'label' => 'Previous'],
+        ['url' => '?page=0', 'label' => '1'],
+        ['url' => '?page=1', 'label' => '2', 'active' => TRUE],
+        ['url' => '?page=2', 'label' => '3'],
+        ['url' => '?page=2', 'label' => 'Next'],
+        ['url' => '?page=7', 'icon' => 'chevron-double-right'],
+      ],
+      7 => [
+        ['url' => '?page=0', 'icon' => 'chevron-double-left'],
+        ['url' => '?page=6', 'label' => 'Previous'],
+        ['url' => '?page=5', 'label' => '6'],
+        ['url' => '?page=6', 'label' => '7'],
+        ['url' => '?page=7', 'label' => '8', 'active' => TRUE],
+      ],
+    ];
+    $argss = [];
+    foreach ($linkss as $page => $links) {
+      $argss[$page] = [
+        [
+          'alignment' => 'center',
+          'links' => $links,
+        ],
+        $page,
+      ];
+    }
+    return $argss;
+  }
+
+  /**
+   * Tests a common pager on different pages.
+   *
+   * @param array $expected_args
+   *   Arguments for a pattern assertion.
+   * @param int $page
+   *   Current page number.
+   *
+   * @dataProvider viewsMiniPagerDataProvider
+   */
+  public function testViewsMiniPager(array $expected_args, int $page): void {
+    \Drupal::requestStack()->push(
+      Request::create('/admin', 'GET', [
+        'page' => $page,
+      ]),
+    );
+
+    /** @var \Drupal\Core\Pager\PagerManagerInterface $pager_manager */
+    $pager_manager = $this->container->get('pager.manager');
+    $pager_manager->createPager(75, 10);
+
+    $html = $this->renderByValue([
+      '#theme' => 'views_mini_pager',
+      '#tags' => [
+        1 => 'The previous',
+        3 => 'The next',
+      ],
+    ]);
+
+    $pagination_assert = new PaginationPatternAssert();
+
+    $pagination_assert->assertPattern($expected_args, $html);
+  }
+
+  /**
+   * Data provider.
+   *
+   * @return array[]
+   *   Argument lists.
+   */
+  public function viewsMiniPagerDataProvider(): array {
+    $linkss = [
+      0 => [
+        ['url' => '', 'label' => '1'],
+        ['url' => '/?page=1', 'label' => 'The next'],
+      ],
+      1 => [
+        ['url' => '/?page=0', 'label' => 'The previous'],
+        ['url' => '', 'label' => '2'],
+        ['url' => '/?page=2', 'label' => 'The next'],
+      ],
+      7 => [
+        ['url' => '/?page=6', 'label' => 'The previous'],
+        ['url' => '', 'label' => '8'],
+      ],
+    ];
+    $argss = [];
+    foreach ($linkss as $page => $links) {
+      $argss[$page] = [
+        [
+          'alignment' => 'center',
+          'links' => $links,
+        ],
+        $page,
+      ];
+    }
+    return $argss;
+  }
+
+  /**
+   * Renders a render element, but does not expect a by-reference parameter.
+   *
+   * @param array $element
+   *   Element to render.
+   *
+   * @return string
+   *   The rendered HTML.
+   */
+  protected function renderByValue(array $element): string {
+    return $this->renderRoot($element);
```

Comment:
It allows inline calls, because it is not by reference.
```php
$element = [...];
render($element);
```
vs
```php
render([...]);
```
But you are free to disagree, maybe it is overkill now. In another version I had more calls to this method.

Reply:
Removed.